client/client.go

```diff
@@ -316,11 +317,83 @@ func makeQueryString(data interface{}) (string, error) {
 	return "", fmt.Errorf("unsupported query string data: %#v", data)
 }
 
+// Remove all custom request serializer logic once APP-1331 is rolled out.
+type serializer func(interface{}) ([]byte, error)
+
+func serializeUnityCatalogDeleteRequest(data interface{}) ([]byte, error) {
+	m := make(map[string]interface{})
+	// Reflectively look up the Force field of data, and if it is true, set the
+	// "force" field of m to true.
```

Comment:
Why didn’t you use query.Values(data) and then converted to map[str]any and then json marshal it?

 It’s hard to understand what the code below is doing.

Reply:
query.Values(data) converts values to `string`, but we want to serialize the request body using the underlying type (e.g. bool for "force").

Reply:
got it!

Reply:
okay, we don't really need `setField` in the current shape - it's too complex. the better solution is something like:

```go
    rv := reflect.ValueOf(data)
    rt := reflect.TypeOf(data)
    for i := 0; i < rv.NumField(); i++ {
        field := rv.Field(i)
        tag := rt.Field(i).Tag.Get("query")
        if tag == "" {
            continue
        }
        value := field.Interface()
        if reflect.IsZero(value) {
           continue
        }
        key := strings.TrimSuffix(tag, ",omitempty") 
        m[key] = value
    }
```